### PR TITLE
commonlib: Update reference links (http -> https)

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- Update Vulnerabilities' references and descriptions to use https links and retire some which were out-dated (Issue 8262).

## Related Issues
Related to: zaproxy/zaproxy#8262

## Checklist
- [NA] Update help
- [X] Update changelog
- [X] Run `./gradlew spotlessApply` for code formatting
- [X] Write tests
- [NA] Check code coverage
- [X] Sign-off commits
- [X] Squash commits
- [X] Use a descriptive title

### CODE REVIEW

**Comment by kingthorin** [comment]
I need to sync refs across files?

**Comment by thc202** [comment]
The tests validate the translated files have the expected number of references, since you are removing/adding it will fail. Better to disable the test and can be enable once they are updated from Crowdin.

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<?xml version="1.0" encoding="utf-8" ?>
-<vulnerabilities xmlns="http://tempuri.org/XMLSchemaOptions.xsd">
+<?xml version="1.0" encoding="utf-8"?>
+<vulnerabilities xmlns="https://tempuri.org/XMLSchemaOptions.xsd">
```
Comment (thc202) [inline addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:2]: HTTPS is not supported.

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml [hunk 2]
```diff
@@ -452,8 +450,8 @@ Pages that provide different responses based on the validity of the data can als
 Servers may include well-known default accounts and passwords. Failure to fully lock down or harden the server may leave improperly set file and directory permissions. Misconfigured SSL certificates and encryption settings, the use of default certificates, and improper authentication implementation with external systems may compromise the confidentiality of information.
 
 Verbose and informative error messages may result in data leakage, and the information revealed could be used to formulate the next level of attack. Incorrect configurations in the server software may permit directory indexing and path traversal attacks.</desc>
-	<solution></solution>
```
Comment (thc202) [inline addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:455]: Revert these.

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml [hunk 3]
```diff
@@ -1069,13 +1081,13 @@ A Web application should invalidate a session after a predefined idle time has p
 6. **Security Headers:** Utilize security headers (e.g., Content Security Policy) to control how browsers and search engines interact with your website.
 
 A proactive and layered approach to security is crucial for safeguarding against Insecure Indexing.</solution>
-	<reference>http://projects.webappsec.org/Insecure-Indexing</reference>
+<reference/>
```
Comment (thc202) [inline addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:1084]: Would be good to keep a reference.

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml (hunk 1 above)
Comment (kingthorin) [reply addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:2]: Oh I didn't even try it, when I saw "tempuri" I figured it was a placeholder

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml (hunk 3 above)
Comment (kingthorin) [reply addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:1084]: I really didn't find a decent resource for this, will look again.

**Comment by kingthorin** [comment]
Thanks will re-adjust.

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml (hunk 1 above)
Comment (kingthorin) [reply addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:2]: Yeah this doesn't seem to exist anymore. According to archive.org it did serve content at some point but now just redirects to Bing also https is offered but not for that DNS the cert offered is for *.oneroute.microsoft.com.

How about we change to https://zaproxy.org/vulnerabilities, it's just a unique name space it doesn't have to actually exist. (From my reading anyway.... 🤷‍♂️)

Comment (thc202) [reply addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml]: Or just remove it? Fine either way though.

Comment (kingthorin) [reply addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml]: Removed.

**Comment by thc202** [comment]
It has conflicts now.

**Comment by kingthorin** [comment]
Deconflicted

**Review by psiinon — APPROVED** [review]

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml [hunk 4]
```diff
@@ -577,7 +577,8 @@ Inputs should be decoded and canonicalized to the application's current internal
 Consider performing repeated canonicalization until your input does not change any more. This will avoid double-decoding and similar scenarios, but it might inadvertently modify inputs that are allowed to contain properly-encoded dangerous content.
 
 When exchanging data between components, ensure that both components are using the same character encoding. Ensure that the proper encoding is applied at each interface. Explicitly set the encoding you are using whenever the protocol allows you to do so.</solution>
-	<reference>http://projects.webappsec.org/Improper-Input-Handling</reference>
+	<reference>https://owasp.org/www-community/vulnerabilities/Improper_Data_Validation</reference>
+	<reference>https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_reference>Sheet.html</reference>
```
Comment (thc202) [inline addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:581]: There's a `reference>` in the middle of the URL.

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml [hunk 5]
```diff
@@ -813,13 +818,13 @@ Even if you make a mistake in your validation (such as forgetting one out of 100
...
 	<solution>Always fully authenticate both ends of any communications channel.
 
 Adhere to the principle of complete mediation.
 
 A certificate binds an identity to a cryptographic key to authenticate a communicating party. Often, the certificate takes the encrypted form of the hash of the identity of the subject, the public key, and information such as time of issue or expiration using the issuer's private key. The certificate can be validated by deciphering the certificate with the issuer's public key. See also X.509 certificate signature chains and the PGP certification structure.</solution>
-	<reference>http://projects.webappsec.org/Routing-Detour</reference>
+	<reference>https://learn.microsoft.com/en-us/previous-versions/ms951244(v=msdn.10)?redirectedfrom=MSDN</reference>
```
Comment (thc202) [inline addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:827]: Remove `?redirectedfrom=MSDN`.

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml [hunk 6]
```diff
@@ -1078,13 +1090,13 @@ A Web application should invalidate a session after a predefined idle time has p
 6. **Security Headers:** Utilize security headers (e.g., Content Security Policy) to control how browsers and search engines interact with your website.
 
 A proactive and layered approach to security is crucial for safeguarding against Insecure Indexing.</solution>
-	<reference>http://projects.webappsec.org/Insecure-Indexing</reference>
+<reference>https://cwe.mitre.org/data/definitions/612.html</reference>
```
Comment (thc202) [inline addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:1093]: Could keep the indentation.

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml [hunk 7]
```diff
@@ -1078,13 +1090,13 @@ A Web application should invalidate a session after a predefined idle time has p
...
+	<desc>Insufficient Password Recovery is when a web site permits an attacker to illegally obtain, change or recover another user's password. Conventional web site authentication methods require users to select and remember a password or passphrase. The user should be the only person that knows the password and it must be remembered precisely. As time passes, a user's ability to remember a password fades. The matter is further complicated when the average user visits 20 sites requiring them to supply a password.  (RSA Survey: https://news.bbc.co.uk/1/hi/technology/3639679.stm) Thus, password recovery is an important part in servicing online users.
```
Comment (thc202) [inline addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:1099]: Does not seem to support `http`.

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml (hunk 4 above)
Comment (kingthorin) [reply addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:581]: Good catch, must have been eclipse trying to auto close the tag or something 🤷‍♂️ will fix.

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml (hunk 7 above)
Comment (kingthorin) [reply addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:1099]: Okay :grinning:

Comment (kingthorin) [reply addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:1099]: If you meant with the "s" then something whack......hmmm checking dev tools it does redirect back to http 🤦‍♂️ 2023

Comment (kingthorin) [reply addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:1099]: I guess we could just drop the in-line reference. The article is amusing but almost 20 years old and doesn't really add that much.

Comment (thc202) [reply addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:1099]: Sounds good to me. (Yeah, had typo in https.)

**Comment by kingthorin** [comment]
Addressed review.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**